impactcommon/exogenous_economy/gdppc.py

```diff
@@ -1,55 +1,213 @@
+        startyear=startyear,
+        stopyear=stopyear
+    )
+    return out
 
-        # Load the growth rates, and split by priority of data
-        df_growth = metacsv.read_csv(files.sharedpath(growth_filepath))
-        df_growth['yearindex'] = np.int_((df_growth.year - baseline_year) / 5)
-        self.df_growth_this = df_growth.loc[(df_growth.model == iam) & (df_growth.scenario == ssp)]
-        self.df_growth_anyiam = df_growth.loc[(df_growth.scenario == ssp)].groupby(['iso', 'year']).median()
-        self.growth_global = df_growth.loc[(df_growth.scenario == ssp) & (df_growth.model == iam)].groupby(['year']).median()
 
-        # Load the nightlights
-        self.df_nightlights = metacsv.read_csv(files.sharedpath(nightlights_filepath))
+def GDPpcProvider(iam, ssp, baseline_year=2010, growth_filepath='social/baselines/gdppc-growth.csv',
+                  baseline_filepath='social/baselines/gdppc-merged-nohier.csv',
+                  nightlights_filepath='social/baselines/nightlight_weight_normalized.csv', stopyear=2100):
+    """Get BestGDPpcProvider through the legacy GDPpcProvider interface
+
+    This interface is deprecated, please use read_bestgdppcprovider() or
+    instantiate BestGDPpcProvider, directly.
+
+    Parameters
+    ----------
+    iam : str
+    ssp : str
+    baseline_year : int, optional
+    growth_filepath : str, optional
+    baseline_filepath : str, optional
+    nightlights_filepath : str, optional
+    stopyear : int, optional
+
+    Returns
+    -------
+    out : impactcommmon.exogenous_economy.BestGDPpcProvider
+
+    Examples
+    --------
+    >>> provider = GDPpcProvider('low', 'SSP3')  # Requires setting IMPERICS_SHAREDDIR.
+    >>> gdppcs = provider.get_timeseries('ZWE.2.2')  # Get the series for any hierid or ISO
+
+    See Also
+    --------
+    read_bestgdppcprovider : Read files on disk to create a BestGDPpcProvider instance
+    BestGDPpcProvider : Provider of GDP per capita (GDPpc) timeseries, selecting "best" data source
+    """
+    warn(
+        "GDPpcProvider is deprecated, please use read_bestgdppcprovider or BestGDPpcProvider, directly",
+        DeprecationWarning
+    )
+    out = read_bestgdppcprovider(
+        iam=iam,
+        ssp=ssp,
+        growth_path_or_buffer=growth_filepath,
+        baseline_path_or_buffer=baseline_filepath,
+        nightlights_path_or_buffer=nightlights_filepath,
+        use_sharedpath=True,
+        startyear=baseline_year,
+        stopyear=stopyear
+    )
+    return out
+
+
+def _get_best_iso_available(iso, df_this, df_anyiam, df_global):
+    """Get the highest priority data available: first data from the IAM, then from any IAM, then global."""
+
+    df = df_this.loc[df_this.iso == iso]
+    if df.shape[0] > 0:
+        return df
+
+    if iso in df_anyiam.index:
+        df = df_anyiam.loc[iso]
+        if df.shape[0] > 0:
+            return df
+
+    return df_global
+
+
+def _split_baseline(df, iam, ssp, baseline_year):
```

Comment:
I don't see the utility in defining these as separate functions, unless you're going to have them provide a reusable service. My suggestion is to do the following:
1. Merge these into a single `_split_hierarchy` function, with logic identical to the current `_split_growth`.
2. Make this a method of `HierarchicalGDPpcProvider`.
3. Use it twice, but when used in the place of `_split_baseline`, first subset by year.

Reply:
Addressed this with `e9182ef2`.

I refactored the two splitting functions into instance methods and removed the tests checking the data splitting.

Your suggested merge doesn't quite produce the same results in the output Dataframe. The "iso" and "year" change position between the column and the index. This means we'd have to make changes elsewhere. Revisit this later once tests are in place?

Reply:
That's fine.